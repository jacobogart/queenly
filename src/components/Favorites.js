import React from "react";

class Favorites extends React.Component {
  constructor() {
    super();
  }

  render() {
    return (
      <div>
        <i className="fas fa-heart fa-2x" />
      </div>
    );
  }
}

export default Favorites;
