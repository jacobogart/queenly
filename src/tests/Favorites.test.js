import React from "react";
import Favorites from "../components/Favorites";
import { shallow } from "enzyme";

const mockResult = {
  "category": "drag show",
  "dayOfWeek": "Sudnay",
  "frequency": "weekly",
  "host": ["Chamilla Foxx"],
  "id": 2101,
  "imageURL": "https://pbs.twimg.com/media/D1Any4VWkAA86HL.jpg",
  "reoccuring": true,
  "startTime": [1400],
  "name": "Sunday Social"
}

const mockDisplayAllSearchResults = jest.fn();
const mockDisplayAllOfType = jest.fn();

describe('Favorites', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      <Favorites
        displayAllSearchResults={mockDisplayAllSearchResults}
        displayAllOfType={mockDisplayAllOfType}
      />
    );
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should call showFavorites on click", () => {
    let faveSpy = jest.spyOn(wrapper.instance(), "showFavorites");
    wrapper.instance().showFavorites();
    expect(faveSpy).toHaveBeenCalled();
    expect(mockDisplayAllSearchResults).toHaveBeenCalled();
    expect(mockDisplayAllOfType).toHaveBeenCalled();
  });
})

