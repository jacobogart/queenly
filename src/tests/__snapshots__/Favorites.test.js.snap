// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Favorites should match snapshot 1`] = `
<div
  className="favoriteBtn"
  onClick={[Function]}
>
  <i
    className="fas fa-heart fa-2x"
  />
</div>
`;
